contessa/consistency_checker.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Dict, Optional
+
+import sqlalchemy
+from datetime import datetime
+
+from contessa.db import Connector
+from contessa.models import create_default_check_class, Table, ResultTable
+
+
+class ConsistencyChecker:
+    """
+    Checks consistency of the sync between two tables.
+    """
+
+    COUNT = "count"
+    DIFF = "difference"
+
+    def __init__(self, left_conn_uri_or_engine, right_conn_uri_or_engine=None):
+        self.left_conn_uri_or_engine = left_conn_uri_or_engine
+        self.left_conn = Connector(left_conn_uri_or_engine)
+        if right_conn_uri_or_engine is None:
+            self.right_conn_uri_or_engine = self.left_conn_uri_or_engine
+            self.right_conn = self.left_conn
+        else:
+            self.right_conn_uri_or_engine = right_conn_uri_or_engine
+            self.right_conn = Connector(right_conn_uri_or_engine)
+
+    def run(
+        self,
+        check_type: str,
+        left_check_table: Dict,
+        right_check_table: Dict,
+        result_table: Dict,
+        context: Optional[Dict] = None,
+    ):
+        left_check_table = Table(**left_check_table)
+        right_check_table = Table(**right_check_table)
+        result_table = ResultTable(**result_table)
+        context = self.get_context(left_check_table, right_check_table, context)
+
+        result = self.do_consistency_check(
+            check_type, left_check_table, right_check_table, context
```

Comment:
this `check_type` confused me with `create_default_check_class(check_type=)`. 
can we rename that? maybe just `type` here would be OK, cause you pass it to consistency checker. what do you think ?

Reply:
i didn't give it a deeper thought, but i like those ideas.

I'd go with `data_quality_dimension` -> as consistency, accuracy are dims of DQ. "quality" is not 😅 but i believe we're gonna get there in the future :)

for the 2nd, i'd go with `method`, it should be readable when - `consistency_checker.run(method="count")`

Reply:
yes, `dimension` is the word I was looking for 😄  changed. I also changed the `CheckType` enum to `DataQualityDimension` for this to be consistent (even though as you pointed out, `quality` is not a dq dimension)